pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -184,22 +183,14 @@ func (s *Server) initK8SConfigStore(args *PilotArgs) error {
 					AddRunFunction(func(leaderStop <-chan struct{}) {
 						// We can only run this if the Gateway CRD is created
 						if configController.WaitForCRD(gvk.KubernetesGateway, leaderStop) {
-							var tagWatcher revisions.TagWatcher
-							// TagWatcher requires permission for MutatingWebhook, so it can't be used in multi-tenant mode
-							if !s.kubeClient.IsMultiTenant() {
-								tagWatcher = revisions.NewTagWatcher(s.kubeClient, args.Revision)
-							}
 							controller := gateway.NewDeploymentController(s.kubeClient, s.clusterID, s.environment,
-								s.webhookInfo.getWebhookConfig, s.webhookInfo.addHandler, tagWatcher, args.Revision)
+								s.webhookInfo.getWebhookConfig, s.webhookInfo.addHandler, args.Revision)
```

Comment:
Will the istio integration tests work if you always remove `tagWatcher`? Shouldn't this be configurable (i.e. we only disable tagWatcher for normal OSSM deployments, but not in integration tests)?

Reply:
> Will the istio integration tests work if you always remove tagWatcher?

Yes.

> Shouldn't this be configurable (i.e. we only disable tagWatcher for normal OSSM deployments, but not in integration tests)?

I thought about it, but it's problematic, because we would have to add yet another argument to istiod, so I think it's not worth to do it.